core/store/src/db/refcount.rs

```diff
@@ -151,14 +179,24 @@ pub(crate) fn iter_with_rc_logic<'a>(
 }
 
 impl RocksDB {
+    /// Merge adds refcounts, zero refcount becomes empty value.
+    /// Empty values get filtered by get methods, and removed by compaction.
+    pub(crate) fn refcount_merge_partial(
```

Comment:
Is the following scenario possible:

- add (key, value1) -- key has rc == 1
- compation with full merge
- del key
- add (key, value2)
- partial merge -> cumulative rc == 0 so value2 is lost
- full merge -> nothing happens and the value at key remains, incorrectly, value1

Reply:
`add` operation (which is the same as refcount increment) disables partial merges entirely. Partial merge operator can only combine two adjacent (for the key) non-positive refcount change operations.

Reply:
(If this scenario you describe is buggy in this PR, it would also be buggy in current implementation...)